### PR TITLE
Declarative setup of Kubevela Addons

### PR DESCRIPTION
**Pull request by ce0la**
- Add Kubevela addons Kubernetes manifests
- Update terragrunt config to deploy it
- Update new cluster code for Kubevela addons

## Summary by CodeRabbit

- **New Features**
  - Introduced dynamic loading of YAML files for managing and adding addons to the application.
  - Added new configuration files for better management of Kubernetes, Helm, and kubectl providers.
  - Updated Terraform source paths for better module management.

- **Refactor**
  - Replaced usage of a specific data source with a more flexible function for file loading.
  - Updated local variable assignments and formatting in configuration files.
  - Changed the source of an environment variable for better configuration management.

- **Chores**
  - Hardcoded a specific value for consistency in the configuration.

### CODE REVIEW

**Comment by ce0la** [comment]
@coderabbitai review

**Comment by renebarbosafl** [comment]
Relates to https://github.com/input-output-hk/sc-dev-platform/pull/32

**Comment by renebarbosafl** [comment]
@ce0la 
I did some changes on infra/modules/kubevela-addons (PR #32) that I think it would be better to get them merged before this PR.

Apparently we don't need to install opentelemetry-operator and its traits because all the Applications are sending traces directly to grafana-agent service. I believe we could add a feature-gate variable to enable/disable it as conveniency.  We don't need of postgres-operator anymore as well.

About KubeVela addons: Are we using o11y-definitions and vela-prism? Where?

I think it would be great if we deploy Traefik as an kubevela Addon instead of deploying it using Terraform module. What do you think? I can help you with this definition.

We also did a lot of breaking changes on this repo and due this I believe we shouldn't mess with prod-us-east-1/k8s right now (As it should be deleted as soon as we migrate everything to the new eks cluster).

File: infra/modules/kubevela-addons/addons.tf [hunk 1]
```diff
@@ -0,0 +1,8 @@
+data "kubectl_filename_list" "manifests" {
+  pattern = "${var.addons_dir}/*.yaml"
+}
+
+resource "kubectl_manifest" "addons" {
+  for_each  = toset(data.kubectl_filename_list.manifests.matches)
+  yaml_body = try(file(each.value), null)
+}
```
[the remark this answers is not in the document]
Comment (ce0la) [reply infra/modules/kubevela-addons/addons.tf]: CoderabbitAI was the one that suggested the current change that uses null. AIs... 🤣

**Review by renebarbosafl — APPROVED** [review]

**Pull request merged**